### DISCUSSION TITLE
[Feature Request] Need A Copy component

### DISCUSSION THREAD

**Discussion by Pszz** [heroui-inc/heroui#4745] [Feature requests]
### Is your feature request related to a problem? Please describe.

I need a Copy button, but it's currently bundled in the <Snippet /> component.

### Describe the solution you'd like

I think the Copy component should be independent, which is more in line with atomicity.

```tsx
<Button isCpoy copyContent="This is the copy content">
    <CopyIcon />
</Button>
```

### Describe alternatives you've considered

See the answer above.

Or...

```tsx
<Copy copyContent="This is the copy content" />
```

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-927/[feature-request]-need-a-copy-component">ENG-927 [Feature Request] Need A Copy component</a></p>

**Comment by abhinav700** [comment]
@wingkwong can you assign this issue to me.

**Comment by GautamPatoliya** [comment]
> @wingkwong can you assign this issue to me.

no. lol 😂

**Comment by wingkwong** [comment]
we haven't decided whether to include this component or not and we don't have the design for that. Hence, will keep it open at this moment.
